src/app/project/project-documents/project-documents.component.ts

```diff
@@ -193,8 +228,13 @@ export class ProjectDocumentsComponent implements OnInit, OnDestroy {
             selBox.style.left = '0';
             selBox.style.top = '0';
             selBox.style.opacity = '0';
-            const safeName = item.documentFileName.replace(/ /g, '_');
-            selBox.value = `${this.pathAPI}/document/${item._id}/fetch/${safeName}`;
+						// Create a fetch link if it's an internal file, provide the link if it's an external file.
+						if (!item.documentFileName.includes('http')) {
+							const safeName = item.documentFileName.replace(/ /g, '_');
+							selBox.value = `${this.pathAPI}/document/${item._id}/fetch/${safeName}`;
+						} else {
+							selBox.value = item.documentFileName;
```

Comment:
```suggestion
							selBox.value = item.externalLink;
```

IMO, it would be more readable to do this instead of assigning `documentFileName` to `externalLink`

Reply:
Ah I see what you're saying. So in the case of the data tables, I mapped all the data before feeding it into the. data table component, so that I gave it uniform values regardless of the source: external link or document. So in this case, I had already mapped the external link to another property because the property externalLink doesn't exist in the format that I feed the table component.